bebraven/platform
# Implement uploading Rise 360 zip as lesson to Canvas module

**NOTES FOR PR**
- Ignore the individual commits. I'll squash for the actual PR. 
- `TODO` ~= Not going to be part of this PR (obviously open to discussion), has Asana task to track

**Task**: https://app.asana.com/0/1174274412967132/1135638232028169

**Description**: A Designer can add a Rise360 package as an item in a module

**Dev environment breaking changes**
- Update your `.env` file with the entries added in `env.example`
- Rebuild for new gems: `docker-compose/scripts/rebuild.sh`
- Migrate for `ActiveStorage`: `deva` then `bin/rails db:migrate`

**Test**
- You'll need to follow the *LTI Extension* set up steps here: https://github.com/bebraven/platform#lti-extension
- Download Rise 360 lesson as zip
- Add lesson to Canvas course > External Tool > LTI Link Extension > Select zip > Add Item
- Click on link and verify it loads
- Screenshots below show sequence of steps (I would have made a video, but the uploading was ~30 seconds of waiting that I didn't know how to edit out):

<img width="1792" alt="Screen Shot 2020-07-15 at 8 27 45 PM" src="https://user-images.githubusercontent.com/62911934/87623735-ed063a80-c6da-11ea-853f-d6e9df2c8cfd.png">
<img width="1792" alt="Screen Shot 2020-07-15 at 8 27 49 PM" src="https://user-images.githubusercontent.com/62911934/87623738-f0012b00-c6da-11ea-8ae5-ed354fa1d509.png">
<img width="1792" alt="Screen Shot 2020-07-15 at 8 27 56 PM" src="https://user-images.githubusercontent.com/62911934/87623740-f1325800-c6da-11ea-8878-cf490270f69e.png">
<img width="1792" alt="Screen Shot 2020-07-15 at 8 28 01 PM" src="https://user-images.githubusercontent.com/62911934/87623743-f42d4880-c6da-11ea-9a81-7ce26e4d0f3a.png">
<img width="1792" alt="Screen Shot 2020-07-15 at 8 29 20 PM" src="https://user-images.githubusercontent.com/62911934/87623746-f5f70c00-c6da-11ea-92b0-935f3a36aab3.png">
<img width="1792" alt="Screen Shot 2020-07-15 at 8 35 40 PM" src="https://user-images.githubusercontent.com/62911934/87623749-f7283900-c6da-11ea-9769-486acb289cd1.png">
<img width="1792" alt="Screen Shot 2020-07-15 at 8 35 44 PM" src="https://user-images.githubusercontent.com/62911934/87623752-f8f1fc80-c6da-11ea-90b4-3482cdacb701.png">
<img width="1792" alt="Screen Shot 2020-07-15 at 8 35 52 PM" src="https://user-images.githubusercontent.com/62911934/87623756-fa232980-c6da-11ea-94d6-3d22768f8a27.png">

Additional ways you can access the course content:
- Clicking on the link you created in the module
- The public object URL for the index on S3: `https://s3.console.aws.amazon.com/s3/buckets/platform-dev-file-uploads/lessons/{lesson_content_zipfile.key}/index.html`
- Your local development endpoint `platformweb/lesson_contents/{lesson_contents_id}` will also redirect you to the above
